admin/adminActionsRestricted.php

```diff
@@ -543,6 +548,49 @@ public function takeForumModerator(array $params)
 		return l_t('This user had their forum moderator status taken.');
 	}
 
+	public function changeEmail(array $params)
+	{
+		global $DB;
+		global $User;
+
+		$userID = (int)$params['userID'];
+		$newEmail = (string)$params['email'];
+
+		if( !isset($params['reason']) || strlen($params['reason'])==0 )
+		{
+			return l_t("Could not change email because no reason was given.");
+		}
+
+		$changeReason = $DB->msg_escape($params['reason']);
+
+		// check if new email is claimed by another account
+		list($result) = $DB->sql_row("SELECT username FROM wD_Users WHERE email = '".$newEmail."'");
+		if (!empty($result)) 
+		{
+			return l_t("This email has already been taken by another user: %s.", $result);
+		}
+
+		// get and store user's old email
+		$oldEmail = $DB->sql_hash("SELECT username, email FROM wD_Users WHERE id = ".$userID);
+
+		if (empty($oldEmail))
+		{
+			return l_t("User id %s has not been found.", $userID);
+		}
+
+		$time = time();
+		$changedBy = $User->username;
+
+		$DB->sql_put(
+			'INSERT INTO wD_EmailHistory (userID, oldEmail, newEmail, date, reason, changedBy) 
+			VALUES ("'.$userID.'", "'.$oldEmail['email'].'", "'.$newEmail.'", "'.$time.'", "'.$changeReason.'", "'.$changedBy.'")'

```

Comment:
userId shouldn't be a string in the table so remove the ' ' around it in the Values section.

Reply:
Same with date/time too I think since you're making that column as an int in the db table